tests/test_tztrout.py

```diff
@@ -0,0 +1,433 @@
+
+    @pytest.mark.parametrize('tz_name', ['PT', 'PACIFIC'])
+    def test_ids_for_tz_name(self, tz_name):
+        pacific_ids = [
+            u'America/Dawson',
+            u'America/Fort_Nelson',
+            u'America/Los_Angeles',
+            u'America/Metlakatla',
+            u'America/Tijuana',
+            u'America/Vancouver',
+            u'America/Whitehorse',
+            u'Canada/Pacific',
+            u'US/Pacific',
+        ]
+        ids = tztrout.tz_ids_for_tz_name(tz_name)
+        assert ids == pacific_ids
+
+    @pytest.mark.parametrize(
+        'phone, result',
+        [('+1 650 333 4444', [-8 * 60]), ('+1 212 333 4444', [-5 * 60])],
+    )
+    def test_non_dst_offsets_for_phone(self, phone, result):
+        offsets = tztrout.non_dst_offsets_for_phone(phone)
+        assert offsets == result
+
+    @pytest.mark.parametrize(
+        'state, result', [('CA', [-8 * 60]), ('NY', [-5 * 60])]
+    )
+    def test_non_dst_offsets_for_address(self, state, result):
+        offsets = tztrout.non_dst_offsets_for_address('US', state=state)
+        assert offsets == result
+
+    @patch('datetime.datetime', FakeDateTime)
+    @pytest.mark.parametrize(
+        'hour_now, range_start, range_end, result',
+        [
+            (
+                20,
+                datetime.time(9),
+                datetime.time(17),
+                [[-11 * 60, -3 * 60], [13 * 60, 14 * 60]],
+            ),
+            (
+                0,
+                datetime.time(9),
+                datetime.time(17),
+                [[9 * 60, 14 * 60], [-14 * 60, -7 * 60]],
+            ),
+            (
+                4,
+                datetime.time(9),
+                datetime.time(17),
+                [[5 * 60, 13 * 60], [-14 * 60, -11 * 60]],
+            ),
+            (7, datetime.time(9), datetime.time(17), [[2 * 60, 10 * 60], ]),
+            (
+                20,
+                datetime.time(17),
+                datetime.time(9),
+                [[-3 * 60, 13 * 60], [-14 * 60, -11 * 60]],
+            ),
+            (0, datetime.time(17), datetime.time(9), [[-7 * 60, 9 * 60], ]),
+            (
+                4,
+                datetime.time(17),
+                datetime.time(9),
+                [[13 * 60, 14 * 60], [-11 * 60, 5 * 60]],
+            ),
+            (
+                7,
+                datetime.time(17),
+                datetime.time(9),
+                [[10 * 60, 14 * 60], [-14 * 60, 2 * 60]],
+            ),
+            (20, '9am', '5pm', [[-11 * 60, -3 * 60], [13 * 60, 14 * 60]]),
+            (20, '9:00', '17:00', [[-11 * 60, -3 * 60], [13 * 60, 14 * 60]]),
+        ],
+    )
+    def test_offset_ranges(self, hour_now, range_start, range_end, result):
+        FakeDateTime.set_utcnow(datetime.datetime(2013, 1, 1, hour_now))
+        offset_ranges = tztrout.offset_ranges_for_local_time(
+            range_start, range_end
+        )
+        assert offset_ranges == result
```

Comment:
It's hard to gauge just by looking at the params and the test code – how does this test work? And what is it asserting?

Reply:
I switched around the variable names to make them clearer, and added a comment for what the test is doing. Let me know if that makes things clearer: 6b3c8ce